iwyu_test_util.py

```diff
@@ -92,6 +137,32 @@ def HasTarget(target):
         supported.append(m.group(1))
     return target in supported
 
+  @staticmethod
+  def UsingCXXstdlib(lib):
+    id_macro = {
+      'libcxx' : '_LIBCPP_VERSION',
+      'libstdcxx' : '__GLIBCXX__',
+      'msvc' : '_MSVC_STL_UPDATE', # https://github.com/microsoft/STL/wiki/Macro-_MSVC_STL_UPDATE
+    }
+    if lib not in id_macro:
+      raise SyntaxError('Unknown C++ standard library: %s' % lib)
+
+    return id_macro[lib] in Features._GetCXXPreprocessorMacros()
+
+  @staticmethod
+  def UsingCstdlib(lib):
+    id_macro = {
+      'glibc' : '__GLIBC__',
+      'macos' : '__APPLE_CC__', # This identifies the Apple compiler rather than libc.
```

Comment:
I guess this will only happen if we've somehow built `include-what-you-use` against the libraries that form the Apple clang compiler. Are those libraries ever shipped? I.e. can this ever happen?

Reply:
Sorry, the comment is incorrect - I clearly wasn't thinking too hard. Given it's name and the documentation I found online, that's what I'd expect it to indicate.
We do actually see this defined on MacOS when either `stddef.h` or `cstddef` are included. I get:
    #define __APPLE_CC__ 6000
    #define __APPLE__ 1
Actually, if I pipe "" into IWYU, I still get these defines, so I think this must be comming from clang's Darwin platform definitions?
Happy to take suggestions for what to comment here. For now I think I'll remove...

Reply:
Yeah, it seems like if the target is any Darwin platform, this will be defined: https://github.com/llvm/llvm-project/blob/main/clang/lib/Basic/Targets/OSTargets.cpp#L25 and https://github.com/search?q=repo%3Allvm%2Fllvm-project%20getDarwinDefines&type=code.

I guess the comment could say: `# Assume Apple libc for any Darwin platform`... or is that too obtuse?

Reply:
Hmm, it looks like `DARWIN_API_VERSION` might be a good detector for Apple libc: https://github.com/apple-open-source-mirror/Libc/blob/5e566be7a7047360adfb35ffc44c6a019a854bea/os/api.h#L74.

Haven't had a chance to test it on an Apple system, but it looks promising.

Reply:
Just checked and I'm not seeing `*DARWIN*` being defined. Looking at the header files on my machine I thought I should be getting `__DARWIN_C_LEVEL` via `sys/cdefs.h`, but I'm not seeing that either (with IWYU or Xcode clang). Not quite sure what's going on.

Reply:
Haven't been able to look at this for a while. I've updated the latest commits with what I've found running on MacOS. Using `__APPLE__` and `__MACH__` appears to be the best option. Would be good to verify that this code identifies `glibc` on a Linux host

Reply:
Hmmm. Looks like switching to a header like `<stdio.h>` allows us to see the system definitions in more cases